src/main/java/net/tropicraft/core/client/ScubaHandler.java

```diff
@@ -97,85 +138,48 @@ public void onTickPlayer(PlayerTickEvent event) {
 			}
 
 		} else {
-		//	d.targetSwimSpeed = 0f;
-		//	d.currentSwimSpeed = 0f;
-			//d.currentRotationPitch = 0;
-			d.currentSwimSpeed *= 0.8f;
-			d.currentRotationYaw = d.rotationYaw;
+			d.targetSwimSpeed = 0f;
 			p.setNoGravity(false);
 		}
 	}
 
 	@SubscribeEvent
 	public void onRenderPlayer(RenderPlayerEvent.Pre event) {
-
-		GlStateManager.pushMatrix();
 		EntityPlayer p = event.getEntityPlayer();
 		PlayerSwimData d = getData(p);
-
+		
 		//
 		//   EntityRenderRegistry.feetModel = new ModelScubaGear(0, EntityEquipmentSlot.FEET);
 		//
 		
-		if (p.isInWater()) {
-			// Set Defaults
-			d.targetRotationYaw = p.rotationYaw;
-			d.targetHeadPitchOffset = 45f;
-			d.targetRotationRoll = (float) p.moveStrafing * 90;
-
-			// Not moving, level out
-			if (p.moveForward == 0f) {
-				d.targetRotationPitch = 0f;
-				d.targetHeadPitchOffset = d.currentSwimSpeed*45;
-			}
-			
-			if(d.targetRotationRoll < 0) {
-				d.targetRotationPitch -= 180f;
-			}
-			d.targetRotationPitch += d.targetRotationRoll;
-			
-
-			// Backpaddle body and head adjustments
-			if (p.moveForward < 0f) {
-				d.targetRotationPitch = p.rotationPitch-65f;
-				d.targetHeadPitchOffset = p.rotationPitch-45f;
-			}
+		updateSwimRenderAngles(p);
+		
+		BlockPos bp = new BlockPos((int) p.posX, (int) p.posY + 2, (int) p.posZ);
```

Comment:
you could probably do BlockPos bp = p.getPos().up(2) here

Reply:
got it